traits/trait_types.py

```diff
@@ -3810,7 +3810,7 @@ def as_ctrait(self):
         return self.trait_maker.as_ctrait()
 
 
-class NoneTrait(TraitType):
+class _NoneTrait(TraitType):
```

Comment:
Could we maybe take the opportunity to note in the docstring that this is primarily for use in the `Either` implementation?

Reply:
Gah. `Union`, not `Either`, of course.

Reply:
:+1: Yes it would!